core/src/main/scala/org/apache/spark/status/protobuf/StageDataWrapperSerializer.scala

```diff
@@ -0,0 +1,627 @@
+    )
+  }
+
+  private def deserializeExecutorMetricsDistributions(
+      binary: StoreTypes.ExecutorMetricsDistributions): ExecutorMetricsDistributions = {
+    new ExecutorMetricsDistributions(
+      quantiles = binary.getQuantilesList.asScala.map(_.toDouble).toIndexedSeq,
+      taskTime = binary.getTaskTimeList.asScala.map(_.toDouble).toIndexedSeq,
+      failedTasks = binary.getFailedTasksList.asScala.map(_.toDouble).toIndexedSeq,
+      succeededTasks = binary.getSucceededTasksList.asScala.map(_.toDouble).toIndexedSeq,
+      killedTasks = binary.getKilledTasksList.asScala.map(_.toDouble).toIndexedSeq,
+      inputBytes = binary.getInputBytesList.asScala.map(_.toDouble).toIndexedSeq,
+      inputRecords = binary.getInputRecordsList.asScala.map(_.toDouble).toIndexedSeq,
+      outputBytes = binary.getOutputBytesList.asScala.map(_.toDouble).toIndexedSeq,
+      outputRecords = binary.getOutputRecordsList.asScala.map(_.toDouble).toIndexedSeq,
+      shuffleRead = binary.getShuffleReadList.asScala.map(_.toDouble).toIndexedSeq,
+      shuffleReadRecords = binary.getShuffleReadRecordsList.asScala.map(_.toDouble).toIndexedSeq,
+      shuffleWrite = binary.getShuffleWriteList.asScala.map(_.toDouble).toIndexedSeq,
+      shuffleWriteRecords = binary.getShuffleWriteRecordsList.asScala.map(_.toDouble).toIndexedSeq,
+      memoryBytesSpilled = binary.getMemoryBytesSpilledList.asScala.map(_.toDouble).toIndexedSeq,
+      diskBytesSpilled = binary.getDiskBytesSpilledList.asScala.map(_.toDouble).toIndexedSeq,
+      peakMemoryMetrics = deserializeExecutorPeakMetricsDistributions(binary.getPeakMemoryMetrics)
+    )
+  }
+
+  private def deserializeExecutorPeakMetricsDistributions(
+      binary: StoreTypes.ExecutorPeakMetricsDistributions): ExecutorPeakMetricsDistributions = {
+    new ExecutorPeakMetricsDistributions(
+      quantiles = binary.getQuantilesList.asScala.map(_.toDouble).toIndexedSeq,
+      executorMetrics = binary.getExecutorMetricsList.asScala.map(
+        ExecutorMetricsSerializer.deserialize(_)).toIndexedSeq
+    )
+  }
+
+  private def deserializeTaskData(binary: StoreTypes.TaskData): TaskData = {
+    val resultFetchStart = getOptional(binary.hasResultFetchStart,
+      () => new Date(binary.getResultFetchStart))
+    val duration = getOptional(binary.hasDuration, () => binary.getDuration)
+    val accumulatorUpdates = AccumulableInfoSerializer.deserializeAccumulableInfos(
+      binary.getAccumulatorUpdatesList)
+    val taskMetrics = getOptional(binary.hasTaskMetrics,
+      () => deserializeTaskMetrics(binary.getTaskMetrics))
+    new TaskData(
+      taskId = binary.getTaskId,
+      index = binary.getIndex,
+      attempt = binary.getAttempt,
+      partitionId = binary.getPartitionId,
+      launchTime = new Date(binary.getLaunchTime),
+      resultFetchStart = resultFetchStart,
+      duration = duration,
+      executorId = weakIntern(binary.getExecutorId),
```

Comment:
When should we use `weakIntern`? Not all serializers use it, will this affect performance?

For example, when `new AccumulableInfo` in `AccumulableInfoSerializer`, we didn't use `weakIntern`

Reply:
For consistency, we use weak here, eg:
https://github.com/apache/spark/blob/c4619b503e58da38c6223020a73c2ca2e0a8c0fa/core/src/main/scala/org/apache/spark/status/protobuf/TaskDataWrapperSerializer.scala#L100-L103

As far as I know, when the field is of type string (not include map<string....>)